### PR TITLE
addressing issue #124

### PR DESCRIPTION
**Pull request by danlu1**
There are several changes in this PR:
1. deletes output_file and output_type arguments, and utilizes the schemas.yml to extract filename and output folder.
2. allow user to decide whether to sort the output as the json file or not.

### CODE REVIEW

File: code/python/create_template_from_Syn_schema.py [hunk 1]
```diff
@@ -12,15 +12,13 @@
...
-                  Full pathname to the output template file
-                  Desired output - either csv or excel
-                  Config_file - /home/ec2-user/sysbioDCCjsonschemas/config/schemas.yml including all registered schemas
-Outputs: csv template files or Excel workbook
-
-Execution: create_template_from_Syn_schema.py <JSON schema name>
-             <output file> <csv/excel> <config_file>
+Input parameters: json_schema_name - Full name of the registered schema, i.e. schema id
+                  config_file - schemas.yml that includes all registered schemas, e.g. /home/ec2-user/sysbioDCCjsonschemas/config/schemas.yml 
+                  template_json - Optional. Specify it when you want to sort the output template as the json file.
```
Comment (avanlinden) [inline code/python/create_template_from_Syn_schema.py:17]:
```suggestion
                  template_json - Optional. Specify it when you want the output template columns to match the order of the terms as they appear in a json file. 
```

**Review by avanlinden — CHANGES_REQUESTED** [review]
I tested this out and it works great, but I think we need an option to specify _which_ config to use, PEC or AD. When I tried updating a shared template like TMTquantitation, it successfully stored to the AD templates folder but threw an error message because I don't have write access to the PEC folder in Synapse.

**Comment by danlu1** [comment]
One clarification question. Do we want the shared template identical to
both AD and PEC?

[image: image.png]
*Dan Lu*(she/her)
Bioinformatics Engineer/Analyst, Sage Bionetworks
Cell: 206-601-6116
http://sagebionetworks.org
<https://na01.safelinks.protection.outlook.com/?url=http%3A%2F%2Fsagebionetworks.org%2F&data=04%7C01%7C%7C14a199bea8bc427c118a08d89b278d60%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C1%7C637429941506870273%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=6EoKb58z5l8vGNR2%2B%2BOQFYDUjjOMdGP6r%2BBhvNcJyx8%3D&reserved=0>

**Comment by avanlinden** [comment]
@danlu1 Yes, if the config yaml is:

```
metadataTemplates.assay-X:
- AD: syn123
- PEC: syn456
```

then the templates should be identical.

**Comment by danlu1** [comment]
Gotcha. Then I think we can give each other the writing permission to the template folder since we'd like to update the shared template all together. I've granted you administrator permission to PEC template folder.

**Pull request merged**